### PR TITLE
regmgr: Move value from string to vector char.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- regmgr: Move value from string to vector char.
reworks GetBin
- Stub sceRegMgrGetKeysInfo.

### CODE REVIEW

File: vita3k/regmgr/src/regmgr.cpp [hunk 1]
```diff
@@ -271,49 +268,57 @@ static void init_system_dreg(RegMgrState &regmgr, const std::wstring pref_path)
 
     for (const auto &reg : reg_template) {
         for (const auto &entry : reg.second) {
-            regmgr.system_dreg[reg.first][entry.name] = entry.value;
+            const std::vector<char> default_value(entry.value.begin(), entry.value.end());
+            auto &value = regmgr.system_dreg[reg.first][entry.name];
+            value.resize(entry.size);
+            value = default_value;
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:274]:
```suggestion
            auto &value = regmgr.system_dreg[reg.first][entry.name];
            value.assign(entry.value.begin(), entry.value.end());
```

File: vita3k/regmgr/src/regmgr.cpp [hunk 2]
```diff
@@ -271,49 +268,57 @@ static void init_system_dreg(RegMgrState &regmgr, const std::wstring pref_path)
...
+    regmgr.system_dreg[fix_category(category)][name].assign(data, data + bufSize);
 
     save_system_dreg(regmgr, pref_path);
 }
 
 int32_t get_int_value(RegMgrState &regmgr, const std::string &category, const std::string &name) {
     std::lock_guard<std::mutex> lock(regmgr.mutex);
 
-    return std::stoi(regmgr.system_dreg[fix_category(category)][name]);
+    const auto sys = regmgr.system_dreg[fix_category(category)][name];
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:297]:
```suggestion
    const auto& sys = regmgr.system_dreg[fix_category(category)][name];
```

File: vita3k/regmgr/src/regmgr.cpp [hunk 3]
```diff
@@ -271,49 +268,57 @@ static void init_system_dreg(RegMgrState &regmgr, const std::wstring pref_path)
...
+
+    const auto str_value = std::to_string(value);
+    regmgr.system_dreg[fix_category(category)][name].assign(str_value.begin(), str_value.end());
 
     save_system_dreg(regmgr, pref_path);
 }
 
 std::string get_str_value(RegMgrState &regmgr, const std::string &category, const std::string &name) {
     std::lock_guard<std::mutex> lock(regmgr.mutex);
+    const auto sys = regmgr.system_dreg[fix_category(category)][name];
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:314]:
```suggestion
    const auto& sys = regmgr.system_dreg[fix_category(category)][name];
```

File: vita3k/regmgr/src/regmgr.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:297]: for why ?? value is just read, no need & 🤔

File: vita3k/regmgr/src/regmgr.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:274]: forget this idea :) i have try is shit and get crazy shit inside file
only works fine like this

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:274]: and ho, it very not like too if you set value inside vector without resize
resize is request before assign too

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:274]: for more explain
that is when i using this code
image
i have no used it for this reason

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: ok is done, sorry i have bad read, but just important, all is empty here, so is obligatory to resize for using assign
if not get big crash

File: vita3k/regmgr/src/regmgr.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: No sur of utility but done

File: vita3k/regmgr/src/regmgr.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: same

File: vita3k/regmgr/src/regmgr.cpp (hunk 2 above)
Comment (saturnsky) [reply vita3k/regmgr/src/regmgr.cpp]: @Zangetsu38 The original code takes extra execution time to allocate memory for the sys variable and copy its value. Macdu's modification uses a reference, which means that the variable will only be read, so it doesn't need to be copied like the original.

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: oh, i see

**Review by Macdu — APPROVED** [review]

**Pull request merged**